src/App.js

```diff
@@ -0,0 +1,272 @@
+import React, { Component } from 'react';
+import { observer } from 'mobx-react'
+import './App.css';
+import { BrowserRouter as Router, Route, Link, Redirect, Switch } from 'react-router-dom'
+import axios from "axios";
+import { AuthService } from './services/Auth.service';
+import Operations from './components/Operations';
+import Transactions from './components/Transactions';
+import Breakdown from './components/Breakdown';
+import Login from './components/Login';
+import 'antd/dist/antd.css';
+import { Layout, Menu, PageHeader, Typography, Space, message } from 'antd';
+import {
+  FileAddOutlined,
+  MoneyCollectOutlined,
+  MenuOutlined,
+  LogoutOutlined
+} from '@ant-design/icons';
+
+const { Sider, Content } = Layout;
+const { Text } = Typography;
+const authService = new AuthService();
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      transactions: [],
+      aggregatedTransactions: [],
+      collapsed: false,
+      redirect: false,
+      loggedIn: authService.isLoggedIn(),
+      pageTitle: 'My Transactions'
+    }
+  }
+
+  setData = async (token, del = false) => {
+    try {
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+      const resp = await axios.get('/transactions', config);
+      const transactions = resp.data.data;
+      const catNames = this.getCategoryNames(transactions);
+      const aggregatedTransactions = await this.getAggrData(catNames);
+      if (del) { return this.setState({ aggregatedTransactions, transactions }, message.success('Transaction deleted', 5)) }
+      this.setState({ aggregatedTransactions, transactions });
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getTransactionsData = async () => {
+    try {
+      const token = authService.getToken()
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+
+      const resp = await axios.get('/transactions', config);
+      const transactions = resp.data.data;
+      const catNames = this.getCategoryNames(transactions);
+      const aggregatedTransactions = await this.getAggrData(catNames);
+
+      return { transactions, aggregatedTransactions }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getAggrData = async (categoryNames) => {
+    const aggrTransactions = []
+    categoryNames.forEach(async (cat) => {
+      const aggr = await this.getAggrCategory(cat);
```

Comment:
You should not use async-await with forEach

Reply:
I see, I'll read on it today and refactor.